packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -34,9 +38,57 @@ function NavigatorProvider(
 		...otherProps
 	} = useContextSystem( props, 'NavigatorProvider' );
 
-	const [ path, setPath ] = useState< NavigatorPath >( {
-		path: initialPath,
-	} );
+	const [ locationHistory, setLocationHistory ] = useState<
+		NavigatorLocation[]
+	>( [
+		{
+			path: initialPath,
+			isBack: false,
+			isInitial: true,
+		},
+	] );
+
+	const push: NavigatorContextType[ 'push' ] = useCallback(
+		( path, options ) => {
+			// Force the `isBack` flag to `false` when navigating forward on both the
+			// previous and the new location.
+			// Also force the `isInitial` flag to `false` for the new location, to make
+			// sure it doesn't get overridden by mistake.
+			setLocationHistory( [
```

Comment:
I think we should add a limit to this array (like 10 items or so)

Reply:
What would be the motivation for it?

Adding a limit would have a few consequences. Let's say the limit is 10 — it means that, when we navigate to the 11th stacked location, we would either:

1. prevent any further navigation. This approach doesn't seem very viable in terms of UX, since it would suddenly prevent the user from navigating to another screen 
2. drop the initial root location, and make the new "first" item in the array the new root. This would introduce further complexity in the code (shifting the array, re-setting the `isInitial` prop, ...) and it would also mean that the user wouldn't be able to navigate back all the way to the initial route.

I personally think that, rather than enforcing a hard limit on the array, we should tackle this potential issue by making sure that our UIs are not designed to require too many levels of "screen nesting" in the first place.

Reply:
Hey @youknowriad , do you have any additional thoughts on my previous answer?

Reply:
Sorry I didn't reply here. I was just thinking about memory leaks but as you said, it might not grow much so fine by me.

Reply:
That's fine, thank you!